### ISSUE TITLE
Allow app compatablility to specified at upload time with addon submission api

### ISSUE THREAD

**Issue by eviljeff** [mozilla/addons-server#18024]
follow on from #7811 - you can specify app compatibility but only the apps themselves (e.g. `"compatibility": ["firefox"]`) are supported.  If a full dict such as `"compatibility": {"firefox": {"min": "61.0", "max": "92.*"}}` was submitted then the extra compatibility data is ignored.

**Comment by eviljeff** [comment]
@ioanarusiczki `compatibility` is a property of `version`, rather than `addon`, so it's being ignored in your JSON (I couldn't replicate your fail - took me a while to notice the subtle difference!).

**Comment by ioanarusiczki** [comment]
@eviljeff Yeap, I already get good results, I'm going to test around and add new results. Thanks!

**Comment by ioanarusiczki** [comment]
@eviljeff 
Thanks for the answers, I've read https://github.com/mozilla/addons-server/issues/7811#issuecomment-943580071 too.

Today's notes:
- I don't see this as an issue but I'll mention : for dicts or lang packs `android` compatibility information can be sent into the request but it is ignored by the response/devhub.  

- the upload API requires the `channel`.  When I set the channel to listed , the response returns at channel : unlisted. 
Same thing when I'm sending a request with channel: unlisted - > the response returns channel: listed. 
This does not seem to affect further the submission with  `api/v5/addons/addon/` or  `v5/addons/addon/620858/versions/` because addons use the channel I requested into `/addons/upload/` -> should I file an issue about it? 

![listed or unlisted](https://user-images.githubusercontent.com/33448286/137736629-028fcd3c-81b7-4dfd-97fd-eaae562c2de4.png)

- I've noticed today that I can use "*" for min and get a 201 response.  The response also returns  `"firefox": {"min": "*", "max": "*"}`
For an extension, dev hub shows an "Incomplete" status, the min versions can be changed by the developer from dev hub but still  I could not approve such extensions even after the min version was set correctly (no Approve button on the listed review pages available). 2 examples : https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620858, https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620860
For a static themes when sending the request with min: * I got a 201, further I approved the theme from rev tools and this is the frontend page, with the warning message that it requires an updated version of FF https://addons-dev.allizom.org/en-US/firefox/addon/oneshot-refuge_1/ -> I think an error should be raised at validation, let me know if I must file this. 

- I filed https://github.com/mozilla/addons-server/issues/18137 for dictionaries.

**Comment by eviljeff** [comment]
> 
> 
> @eviljeff Thanks for the answers, I've read [#7811 (comment)](https://github.com/mozilla/addons-server/issues/7811#issuecomment-943580071) too.
> 
> Today's notes:
> 
>     * the upload API requires the `channel`.  When I set the channel to listed , the response returns at channel : unlisted.
>       Same thing when I'm sending a request with channel: unlisted - > the response returns channel: listed.
>       This does not seem to affect further the submission with  `api/v5/addons/addon/` or  `v5/addons/addon/620858/versions/` because addons use the channel I requested into `/addons/upload/` -> should I file an issue about it?

yes please - I think it's a problem with the serializer (i.e. the response) but I agree it should be fixed.

>     * I've noticed today that I can use "*" for min and get a 201 response.  The response also returns  `"firefox": {"min": "*", "max": "*"}`
>       For an extension, dev hub shows an "Incomplete" status, the min versions can be changed by the developer from dev hub but still  I could not approve such extensions even after the min version was set correctly (no Approve button on the listed review pages available). 2 examples : https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620858, https://reviewers.addons-dev.allizom.org/en-US/reviewers/review/620860
>       For a static themes when sending the request with min: * I got a 201, further I approved the theme from rev tools and this is the frontend page, with the warning message that it requires an updated version of FF https://addons-dev.allizom.org/en-US/firefox/addon/oneshot-refuge_1/ -> I think an error should be raised at validation, let me know if I must file this.

`*` shouldn't be allowed as a valid version number for `min` - I'm going to fix this with https://github.com/mozilla/addons-server/issues/18073
